pydantic/exceptions.py

```diff
@@ -1,37 +1,58 @@
-import inspect
 import json
+from typing import Iterable, Type
 
 from .utils import to_snake_case
 
 __all__ = (
     'Error',
     'ValidationError',
     'ConfigError',
-    'Missing',
-    'Extra',
+
+    'ValueError_',
+    'TypeError_',
+
+    'MissingError',
+    'ExtraError',
+
+    'DecimalError',
+    'DecimalIsNotFiniteError',
+    'DecimalMaxDigitsError',
+    'DecimalMaxPlacesError',
+    'DecimalWholeDigitsError',
+
+    'UUIDError',
+    'UUIDVersionError',
 )
 
 
 class Error:
-    __slots__ = 'exc_info', 'loc'
+    __slots__ = 'exc', 'loc'
 
     def __init__(self, exc, *, loc):
-        self.exc_info = exc
+        self.exc = exc
         self.loc = loc if isinstance(loc, tuple) else (loc,)
 
+    @property
+    def ctx(self):
+        return getattr(self.exc, 'ctx', None)
+
     @property
     def msg(self):
-        return str(self.exc_info)
+        return str(self.exc)
 
     @property
     def type_(self):
-        bases = []
-        for b in inspect.getmro(type(self.exc_info)):
-            bases.append(b.__name__)
-            if b in (ValueError, TypeError):
-                break
+        return get_exc_type(self.exc)
 
-        return to_snake_case('.'.join(bases[::-1]))
+    def as_dict(self, *, loc_prefix=None):
+        loc = self.loc if loc_prefix is None else loc_prefix + self.loc
+
+        return {
+            'loc': loc,
+            'msg': self.msg,
+            'type': self.type_,
+            'ctx': self.ctx,
```

Comment:
Why? For me it's better to return all keys with useful defaults, like `None` if there are no context. In this case user see that `ctx` is possible, but is empty now. Also from my point of view, better to write code like `ctx = error.dict()['ctx']` rather than `ctx = error.dict().get('ctx')`.

Reply:
I think it's better to keep errors respones as short as possible. I understand your point of you but I'd still rather removing `ctx` if it's `None`

Reply:
No problem, fixed.